soc/arm/nordic_nrf/Kconfig.defconfig

```diff
@@ -26,4 +26,9 @@ config ENTROPY_NRF5_RNG
 
 endif # ENTROPY_GENERATOR
 
+config GPIO
+
+config SPI
+	select GPIO
```

Comment:
This will force the inclusion of the GPIO driver with no possibility to prevent it. And this may be undesirable (one may want to handle GPIOTE interrupts in some custom way, for instance).
The GPIO driver is not always required when SPI is used on nRF chips:
- in slave mode it is not needed at all
- in master mode the CS pin is optional and one may decide to not use it (for example, when there is only one slave on the bus and it can be always active or it does not even have the CS input)

Therefore, I think it would be better to only select the GPIO driver by default when SPI is activated (as this will be convenient to have it done automatically) and keep the possibility of deselecting it:
```
if SPI

config GPIO
	default y

endif # SPI
```

Reply:
Agreed; done.